camel/toolkits/linkedin_toolkit.py

```diff
@@ -0,0 +1,288 @@
+
+        access_token = self._get_access_token()
+        headers = {
+            "Authorization": f"Bearer {access_token}",
+            "Content-Type": "application/json",
+        }
+
+        response = requests.delete(
+            f"https://api.linkedin.com/v2/ugcPosts/{post_id}",
+            headers=headers,
+        )
+
+        if response.status_code != HTTPStatus.NO_CONTENT:
+            error_type = self._handle_http_error(response)
+            return (
+                "Request returned a(n) "
+                + str(error_type)
+                + ": "
+                + str(response.status_code)
+                + " "
+                + response.text
+            )
+
+        return f"Post deleted successfully. Post ID: {post_id}."
+
+    def get_my_profile(self) -> str:
+        r"""Retrieves and formats the authenticated user's LinkedIn
+        profile info.
+
+        This function sends a GET request to the LinkedIn API to retrieve the
+        authenticated user's profile information.
+        It then formats this information into a readable report.
+
+        Returns:
+            str: A formatted report of the authenticated user's
+            LinkedIn profile information. This includes their ID,
+            first name, last name, headline,and profile picture.
+            If the request to the LinkedIn API is not successful,
+            the return is an error message.
+
+        Reference:
+            https://docs.microsoft.com/en-us/linkedin/shared/references/v2/profile/basic-profile
+        """
+        access_token = self._get_access_token()
+        headers = {
+            "Authorization": f"Bearer {access_token}",
+            'Connection': 'Keep-Alive',
+            'Content-Type': 'application/json',
+            "X-Restli-Protocol-Version": "2.0.0",
+        }
+
+        response = requests.get(
+            "https://api.linkedin.com/v2/userinfo",
+            headers=headers,
+        )
+
+        if response.status_code != HTTPStatus.OK:
+            error_type = self._handle_http_error(response)
+            return (
+                "Request returned a(n) "
+                + str(error_type)
+                + ": "
+                + str(response.status_code)
+                + " "
+                + response.text
+            )
+
+        json_response = response.json()
+
+        locale = json_response.get('locale', {})
+        country = locale.get('country', 'N/A')
+        language = locale.get('language', 'N/A')
+
+        profile_report = (
+            f"Country: {country}. "
+            f"Language: {language}. "
+            f"First Name: {json_response.get('given_name')}. "
+            f"Last Name: {json_response.get('family_name')}. "
+            f"Email: {json_response.get('email')}. "
+        )
+
+        return profile_report
+
+    def get_my_profile_id(self) -> str:
+        r"""Retrieves the authenticated user's LinkedIn profile ID.
```

Comment:
this method is making the same API call as `get_my_profile`, we can refactor to make these 2 methods into one and return a tuple